code/modules/modular_computers/computers/item/modular_computer.dm

```diff
@@ -0,0 +1,805 @@
+		if(portable_drive)
+			user << "This computer's portable drive slot is already occupied by \the [portable_drive]."
+			return
+		found = 1
+		portable_drive = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/hard_drive))
+		if(hard_drive)
+			user << "This computer's hard drive slot is already occupied by \the [hard_drive]."
+			return
+		found = 1
+		hard_drive = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/network_card))
+		if(network_card)
+			user << "This computer's network card slot is already occupied by \the [network_card]."
+			return
+		found = 1
+		network_card = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/nano_printer))
+		if(nano_printer)
+			user << "This computer's nano printer slot is already occupied by \the [nano_printer]."
+			return
+		found = 1
+		nano_printer = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/card_slot))
+		if(card_slot)
+			user << "This computer's card slot is already occupied by \the [card_slot]."
+			return
+		found = 1
+		card_slot = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/battery_module))
+		if(battery_module)
+			user << "This computer's battery slot is already occupied by \the [battery_module]."
+			return
+		found = 1
+		battery_module = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/processor_unit))
+		if(processor_unit)
+			user << "This computer's processor slot is already occupied by \the [processor_unit]."
+			return
+		found = 1
+		processor_unit = H
+	if(found)
+		user << "You install \the [H] into \the [src]"
+		H.holder2 = src
+		if(!user.drop_item(H))
+			return
+		H.forceMove(src)
+
+// Uninstalls component. Found and Critical vars may be passed by parent types, if they have additional hardware.
+/obj/item/modular_computer/proc/uninstall_component(var/mob/living/user, var/obj/item/weapon/computer_hardware/H, var/found = 0, var/critical = 0)
+	if(portable_drive == H)
+		portable_drive = null
+		found = 1
+	if(hard_drive == H)
+		hard_drive = null
+		found = 1
+		critical = 1
+	if(network_card == H)
+		network_card = null
+		found = 1
+	if(nano_printer == H)
+		nano_printer = null
+		found = 1
+	if(card_slot == H)
+		card_slot = null
+		found = 1
+	if(battery_module == H)
+		battery_module = null
+		found = 1
+	if(processor_unit == H)
+		processor_unit = null
+		found = 1
+		critical = 1
+	if(found)
+		if(user)
+			user << "You remove \the [H] from \the [src]."
+		H.forceMove(get_turf(src))
+		H.holder2 = null
+	if(critical && enabled)
+		if(user)
+			user << "<span class='danger'>\The [src]'s screen freezes for few seconds and then displays an \"HARDWARE ERROR: Critical component disconnected. Please verify component connection and reboot the device. If the problem persists contact technical support for assistance.\" warning.</span>"
+		shutdown_computer()
+		update_icon()
+
+
+// Checks all hardware pieces to determine if name matches, if yes, returns the hardware piece, otherwise returns null
```

Comment:
This isn't the only find_hardware_by_name proc either. Not a good idea to do it this way.